### PR TITLE
Send logrus fields to Bugsnag as metadata.

### PR DESCRIPTION
**Pull request by Roxot**
This adds all `logrus.Fields{ ... }` except for the `"error"` field as Bugsnag metadata, viewable in Bugsnag under the metadata tab.

screen shot 2016-04-05 at 2 54 33 pm

cc @csfrancis

### CODE REVIEW

File: bugsnag.go [hunk 1]
```diff
@@ -53,8 +53,16 @@ func (hook *bugsnagHook) Fire(entry *logrus.Entry) error {
...
+	metadata["metadata"] = make(map[string]interface{})
+	for key, val := range entry.Data {
+		if key != "error" {
+			metadata["metadata"][key] = val
+		}
+	}
+
 	errWithStack := bugsnag_errors.New(notifyErr, skipStackFrames)
-	bugsnagErr := bugsnag.Notify(errWithStack)
+	bugsnagErr := bugsnag.Notify(errWithStack, metadata)
```
Comment (csfrancis) [inline bugsnag.go]: Does it make sense to pass the `metadata` parameter even if it's empty?

File: bugsnag_test.go [hunk 2]
```diff
@@ -23,15 +23,20 @@ type exception struct {
...
+}
+
 type notice struct {
-	Events []struct {
-		Exceptions []exception `json:"exceptions"`
-	} `json:"events"`
+	Events []event `json:"events"`
 }
 
 func TestNoticeReceived(t *testing.T) {
```
Comment (csfrancis) [inline bugsnag_test.go]: If we decide to always pass the `metadata` param to `Notify`, I don't really see the need to test the case where `metadata` is empty. I'd just have one test that also validates the metadata values.

File: bugsnag.go (hunk 1 above)
Comment (Roxot) [inline bugsnag.go]: On Bugsnag itself the metadata tab won't show up if you send it an empty metadata map. It might add a slight overhead to the request, so I could make a check for it if you'd like.

File: bugsnag_test.go (hunk 2 above)
Comment (Roxot) [inline bugsnag_test.go]: Agreed that we're missing a test confirming the metadata values.

File: bugsnag.go (hunk 1 above)
Comment (csfrancis) [inline bugsnag.go]: I don't feel super strongly either way. You can just leave it as-is.

File: bugsnag_test.go (hunk 2 above)
Comment (csfrancis) [inline bugsnag_test.go]: I'd remove the code that tests the case where the metadata map is empty and add the checks for the values.

**Comment by csfrancis** [comment]
LGTM.
